### PR TITLE
Seed and choices fix and bench

### PR DESCRIPTION
**Pull request by satyaog**

### CODE REVIEW

File: src/orion/algo/ax/axoptimizer.py [hunk 1]
```diff
@@ -31,20 +31,42 @@ class AxOptimizer(BaseAlgorithm):
...
-                 random_seed: Optional[int] = None,
                  early_stopping_strategy: Optional["BaseEarlyStoppingStrategy"] = None,
                  choose_generation_strategy_kwargs: Optional[Dict[str, Any]] = None):
+        self._client_state = None
+        self._original_space = copy.deepcopy(space)
+        self._trials_map = {}  # tmp
+
+        for dimension in space.values():
+            if dimension.prior_name == "choices":
+                dimension.categories = tuple(str(c) for c in dimension.categories)
```
Comment (satyaog) [inline src/orion/algo/ax/axoptimizer.py:45]: Would this be a fix to support multi-types in choices?

Comment (bouthilx) [reply src/orion/algo/ax/axoptimizer.py:45]: No because it would change the values defined by the user. It would need to be a set of `transform(orion_params, space)`/`reverse(ax_params, space)` that converts to str in transform, and transform back to original values in reverse.

Comment (bouthilx) [reply src/orion/algo/ax/axoptimizer.py:45]: This way Ax works with values that are compliant with it, but the users still see original categories when trials are suggested (non string).

File: src/orion/algo/ax/axoptimizer.py [hunk 2]
```diff
@@ -153,20 +171,40 @@ def is_done(self):
...
+                for _c in dimension.interval():
+                    if c == str(_c):
+                        if isinstance(_c, int):
+                            result[dimension.name] = int(c)
+                        elif isinstance(_c, float):
+                            result[dimension.name] = float(c)
+                        # else this is a string
+                        break
+
+        return result
```
[the remark this answers is not in the document]
Comment (satyaog) [reply src/orion/algo/ax/axoptimizer.py:192]: Yes it's to differentiate a `None` from a `"None"`

File: tests/benchmark/AxOptimizer.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+pool_size: 1
+max_trials: 5
+
+algorithms:
+  AxOptimizer:
+    seed: 1234
+    generation_strategy: None
+    enforce_sequential_optimization: True
+    early_stopping_strategy: None
+    choose_generation_strategy_kwargs: None
```
Comment (satyaog) [inline tests/benchmark/AxOptimizer.yaml:12]: Is this file needed for `tests/benchmark.py`?

Comment (bouthilx) [reply tests/benchmark/AxOptimizer.yaml:12]: No it should not.

**Review by bouthilx — APPROVED** [review]
